libs/parallelism/algorithms/include/hpx/parallel/datapar/zip_iterator.hpp

```diff
@@ -37,10 +37,11 @@ namespace hpx { namespace parallel { namespace util { namespace detail {
             hpx::util::zip_iterator<Iter...> const& it,
             hpx::util::index_pack<Is...>)
         {
+            if (!sizeof...(Is))
+                return true;
             auto const& t = it.get_iterator_tuple();
-            bool const sequencer[] = {
-                false, is_data_aligned(hpx::get<Is>(t))...};
-            return std::any_of(&sequencer[1], &sequencer[sizeof...(Is) + 1],
+            bool const sequencer[] = {is_data_aligned(hpx::get<Is>(t))...};
```

Comment:
Some compilers might still complain about zero-sized arrays if the sequence happens to be empty. I'd suggest not to add the `if(!sizeof...(Is)) ...` above, but write the sequencer as `= {true, is_data_aligned(...)...}` instead.